### ISSUE TITLE
[Bug] cannot use some names in imports

### ISSUE THREAD

**Issue by oxymc** [storybookjs/addon-svelte-csf#167]
> [!NOTE]
> This issue as a possible [workaround](https://github.com/storybookjs/addon-svelte-csf/issues/167#issuecomment-2104476827)

### Describe the bug

Can't use:
`import { Menu } from './index';` in `index.stories.svelte`
(but it works, if 'Menu' will be changed by other word: 'DropdownMenu' for example)

### Steps to reproduce the behavior

1. Create a new sveltekit app: `npm create svelte@latest <name-of-app>`
2. Now add storybook `npx storybook init`
3. Now add Svelte CSF
4. Reproduce the issue

### Expected behavior

More details are in screenshots

### Screenshots and/or logs

<img width="992" alt="Знімок екрана 2024-01-29 о 09 15 32" src="https://github.com/storybookjs/addon-svelte-csf/assets/59724861/fc338aa9-c3c3-41cc-90d9-7fad9176e4e5">
<img width="589" alt="Знімок екрана 2024-01-29 о 09 18 06" src="https://github.com/storybookjs/addon-svelte-csf/assets/59724861/c15274fc-932e-41c1-935b-155961955c31">

### Environment

 - OS: macOS Sonoma 14.1.2
 - Node.js version: v18.17.1
 - NPM version: 10.0.0
 - Browser (if applicable): Chrome
 - Browser version (if applicable): 121.0.6167.85
 - Device (if applicable): MacBook Pro 2019

### Additional context

Will be good to know which words are 'forbidden' too, but will be great to fix it :)

**Comment by j3rem1e** [comment]
Do you have a repo where I can look ?

Import should be analyzed with a warning/Reallocation of the story with the same name.

**Comment by basselalsayed** [comment]
For me the issue comes with import "conflicts"

```
// menu.svelte

import { Menu } from './menu-base.svelte';

=> docgen error, object is not extensible
```



if we rename to 

```
import { Menu as Base } from './menu-base.svelte';
```

it works. 

I wasn't sure if the error came from this library, storybook, vite or esm lol

**Comment by JReinhold** [comment]
@basselalsayed can you share a repository that demonstrates this problem?

**Comment by basselalsayed** [comment]
@JReinhold yeah here you go https://github.com/basselalsayed/svelte-storybook-csf-issue

in select.svelte, try renaming the import and it works fine

**Comment by ACERposS** [comment]
I am experiencing a similar issue with shadowing of the global `Error` class when mocking GraphQL test cases. The workaround is to use `global.Error`, as in-

    <APIProvider mocks={{
      Query: {
        getRecords: () => { throw new global.Error("Request failure"); },
      },
    }}>
      <ComponentUnderTest />
    </APIProvider>

Perhaps a general pass to reduce name collisions over all the scoped variables in the template would be good, incl. underscore prefixing etc.

**Comment by JReinhold** [comment]
Closing as duplicate of #250
